calyx-opt/src/analysis/live_range_analysis.rs

```diff
@@ -862,7 +862,15 @@ impl LiveRangeAnalysis {
             })
             .collect();
 
-        let comp_is_written = !inputs.is_empty()
+        let written_in_group =
+            comb_group_info.as_ref().map_or(false, |comb_group| {
```

Comment:
I think you can just do `.map(...).unwrap_or_default()` here which will return `false` if the optional is empty.